include/cadmium/core/simulation/parallel_root_coordinator.hpp

```diff
@@ -30,185 +30,240 @@
 #include "root_coordinator.hpp"
 #include "../logger/logger.hpp"
 
+#include <iostream>
+
 namespace cadmium {
     //! Parallel Root coordinator class.
     class ParallelRootCoordinator: public RootCoordinator {
+     private:
+        //! It serializes the IC couplings in pairs <port_to, {ports_from}> to parallelize message propagation.
+        std::vector<std::pair<std::shared_ptr<PortInterface>, std::vector<std::shared_ptr<PortInterface>>>> stackedIC;
      public:
-        ParallelRootCoordinator(std::shared_ptr<Coupled> model, double time):
-            RootCoordinator(std::move(model), time) {}
-        explicit ParallelRootCoordinator(std::shared_ptr<Coupled> model): ParallelRootCoordinator(std::move(model), 0) {}
+        ParallelRootCoordinator(std::shared_ptr<Coupled> model, double time): RootCoordinator(std::move(model), time), stackedIC() {
+            auto coupled = topCoordinator->getCoupled();
+            coupled->flatten();  // In parallel execution, models MUST be flat
```

Comment:
We have a problem here: when we do the flattening process, the abstract simulator has already been created on the constructor. The flattening affects the model, but not the coordinators and simulators already created. And we can't use the flatten function here before calling the RootCoordinator constructor.
As far as I can see the options we have are:

1. Leave it as it was, applying the flattening function before instantiating ParallelRootCoordinator (but the users have the option of doing it or not, even they must to use the parallel version).

2. Redefine the abstract simulator structure calling a function that does the same as the constructor.

@romancardenas what do you think?

Reply:
I think that the best approach here is composition instead of inheritance.
The `ParallelRootCoordinator` would not inherit from `RootCoordinator`. Instead, it will have a `RootCoordinator` attribute. In this way, we can flatten the model in the constructor and then create the `RootCoordinator` with the model already flattened. Perhaps we need to add `ParallelRootCoordinator` to the `friend` list of `RootCoordinator`, but I think this is the best approach. What do you think @gtrabes  ?

Reply:
Agreed and done!